bin/flutter.bat

```diff
@@ -86,11 +86,11 @@ GOTO :after_subroutine
   IF NOT EXIST "%stamp_path%" GOTO do_snapshot
   SET /P stamp_value=<"%stamp_path%"
   IF !stamp_value! NEQ !revision! GOTO do_snapshot
-  REM Compare "last modified" timestamps
+  REM Compare created timestamps
   SET pubspec_yaml_path=%flutter_tools_dir%\pubspec.yaml
   SET pubspec_lock_path=%flutter_tools_dir%\pubspec.lock
-  FOR %%i IN ("%pubspec_yaml_path%") DO SET yaml_timestamp=%%~ti
-  FOR %%i IN ("%pubspec_lock_path%") DO SET lock_timestamp=%%~ti
+  FOR /F "TOKENS=1-2" %%i IN ('DIR /TC "%pubspec_yaml_path%" ^| FINDSTR "^[0-9]"') DO SET yaml_timestamp=%%i %%j
```

Comment:
Cool! 
But what about PM/AM suffix (token=3)?
Also is space-delimited format guaranteed for `dir /tc` date/time output on all locales, is there any way to check?

Reply:
Good catch, whoops. Updated to get AM/PM.

I couldn't find any guarantee of the delimiter, or mention of language changing the output format. I tried changing my locale to spot check it, and didn't see it make a difference. Looks like the format of the date does change to match whatever the preferred "short date format" is in the system settings. But the time output is always consistently `hh:mm t` regardless of preferred time. That should be good since recency is determined with  `dir /o:d`, not by trying to parse the timestamp, and all of the preferred date settings use either `/` or `-` as a delimiter.

Reply:
Scanning through Date/time format settings I see `H:mm`, `h:mm tt`, `h:mmtt`, `tt 'ga' h:mm` all acceptable possible short time formats on Windows for different locales.

Reply:
Were you seeing those as optional preferred short time formats or as the defaults? Which locales?  I wasn't seeing changing the preferred time have any effect, but if the actual default output from `dir` changes that'll cause a problem.

Reply:
Default short time format for "Hungarian (Hungary)", "Ibibio (Nigeria)", "Ewe (Ghana)".